### DISCUSSION TITLE
Save Sessions during update

### DISCUSSION THREAD

**Discussion by daviddr17** [devlikeapro/waha#455] [General]
Maybe I'm missing something, but how do I save all sessions during an update so that I don't have to create all sessions again after the update and therefore don't have to rescan the codes?

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
Hi! 
You need to explicitly enable one of the storages:
https://waha.devlike.pro/docs/how-to/storages/

Here's a simple file storage (but works perfectly even with 400+ session in production)
```
docker run -v `pwd`/.sessions:/app/.sessions -p 3000:3000/tcp devlikeapro/waha-plus
```

All session data will be save in `.sessions` folder outside of the container.

Consider using docker-compose to run it instead of manually putting all data in docker run command, here's example:

https://github.com/devlikeapro/waha/blob/core/docker-compose.yaml#L18-L45
1. stop previous container
2. Copy file
3. Remove all services but `waha-plus`
4. Change values
5. run `docker-compose up -d`
6. open http://localhost:3000

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
